### ISSUE TITLE
Service file and directory permissions

### ISSUE THREAD

**Issue by jsirex** [habitat-sh/habitat#6076]
```bash
# ls -lha /hab/svc/builder-minio/
total 44K
drwxr-xr-x 9 root root 4.0K Jan 23 17:27 .
drwxr-xr-x 3 root root 4.0K Jan 22 14:52 ..
-rw-r--r-- 1 root root    4 Jan 23 17:27 PID
drwxrwx--- 3 hab  hab  4.0K Jan 22 14:52 config
drwxrwx--- 5 hab  hab  4.0K Jan 23 17:27 data
drwxrwx--- 2 hab  hab  4.0K Jan 22 14:52 files
drwxr-xr-x 2 root root 4.0K Jan 23 17:27 hooks
drwxr-xr-x 2 root root 4.0K Jan 23 17:21 logs
-rwxr-xr-x 1 root root 1.8K Jan 23 17:27 run
drwxrwx--- 2 hab  hab  4.0K Jan 23 12:52 static
drwxrwx--- 2 hab  hab  4.0K Jan 22 14:52 var
```

`hooks`, `logs` and `run` script may contain sensitive data:
```bash
find core-plans -type f -wholename '*/hooks/*' -exec grep -EHin 'user|pass|token|secret|login' {} \; | wc
    109     615   10970
```
Probably it should have owned by `hab` and `0770` mode.

**Comment by christophermaier** [comment]
What version of the Supervisor are you using? What is the fully-qualified package identifier for the `builder-minio` service you're using?

I'd have expected these to be owned by `hab` for the `core/builder-minio` service.

**Comment by jsirex** [comment]
I actually tested on `habitat/builder-minio` - from my PR. It merged, so I used https://github.com/habitat-sh/builder/tree/master/components/builder-minio

But there aren't any special chown/chmod commands. 
I used `core/hab-sup/0.73.0/20190115010906` and `core/hab-launcher/9167/20181203215950/bin/hab-launch`

**Comment by christophermaier** [comment]
Sorry, I meant `habitat/builder-minio`... old habits die hard :)

Thanks for the extra info; we'll take a look at this!

**Comment by jsirex** [comment]
I've found new issue related to this with files permissions and now it is critical and many services which use last habitat may fail:
Supervisor renders pid file with `0600` mode. So if you're running service under different user than `root` you cannot read PID file

**Comment by jsirex** [comment]
UP: This permissions issue get annoying but seems trivial to fix, doesn't it?

**Comment by stevendanna** [comment]
I agree, let's see if we can sort it out. I think the proposal is that 

`hooks` -> owned by svc_user:svc_group with mode `0770`
`logs` -> owned by svc_user:svc_group with mode `0770`
`run` -> owned by svc_user:svc_group with mode `0770`
`PID` -> owned by hab-sup's user (root) with mode `0644`

The PID file permission was a recent change caused by the AtomicWriter, let's track that in the other bug here: https://github.com/habitat-sh/habitat/issues/7351

For the hooks and logs directory, it appears that the root ownership of these directories was at least partially intentional:

https://github.com/habitat-sh/habitat/blob/master/components/core/src/fs.rs#L384-L391

My _guess_ is that at least part of the story here is that people didn't want `svc_user` owning these directories because they didn't want to _require_ that hab-sup run as root and hab-sup needs to be able to write to those files/directories.  Right now _technically_ you could run hab-sup as a non-root user but with CAP_SETUID, CAP_SETGID, and CAP_CHOWN and things will work (or at least the code thinks it'll work). This then leads to read and execute permissions for other on those directories and files in case the service itself wants to be able to read them.

Perhaps some people on the core team can confirm if this is the main use case that we need to work around.

**Comment by christophermaier** [comment]
@stevendanna Your analysis lines up with my recollection of things. All the `CAP_*` stuff came in to enable the Supervisor to be able to run as a non-root user, which was important for being able to get it running in OpenShift (see [this blog post](https://blog.chef.io/habitat-can-now-run-in-openshift/) and the corresponding [docs](https://www.habitat.sh/docs/best-practices/#running-habitat-linux-containers) for further background).

The PID issue probably does need to be fixed, but only for backwards-compatibility issues, because the latest Supervisor by default does not use PID files for services (this in turn depends on using the latest stable Launcher).

**Comment by jsirex** [comment]
I'm less thinking about `CAP_CHOWN` and possibility to set correct owner (usually files are created by correct owner), but more worried about security issue: hooks, run, logs are world readable and may contain sensitive information

**Comment by stevendanna** [comment]
> https://github.com/habitat-sh/habitat/issues/6076#issuecomment-588484861

Yup, I agree! The comments above are just working through what the impact of various permission and ownership changes are.

(1) Having these directories/files owned by the hab_svc user will impact the ability for hab-sup to run as a non-root user.

(2) Having these directories/files be mode 0770 but owned by root I _think_ will break anyone who is doing something like sourcing one hook from another hook while running their service as a non-root user.

I'm thinking the following might be good:

Mode: 0770 or 0750
Owernship: hab-sup user, hab-svc GROUP

This would allow the svc_user to still _read_ the files in those directories via the group permissions, but disallow the world readability.

**Comment by jsirex** [comment]
5 cents on PID. Now, when new launcher doesn't create a new PID file how can I get services' PID from `reconfigure` hook?

**Comment by christophermaier** [comment]
@jsirex Hrmm... there's an oversight :man_facepalming: For the time being, using `core/hab-launcher/12605/20191112144831` or earlier will continue using PID files. I'll write up an issue to address this later today.
